### ISSUE TITLE
DataSet ReadXmlSchema Error when uri filename contain special character

### ISSUE THREAD

**Issue by pulkjr** [dotnet/corefx#28960]
When running .net core on mac was unable to pass string with special character to ReadXmlSchema, specifically a '#'. The error return path not found stopping the given path before the special character. Workaround using %23 works, but using '#' on windows system works fine. Attempted to use %23 on windows system and was unable to resolve path. Appears to be error coming from XmlTextReaderImpl or XmlResolver. Using uri and tostring method returns correct path, '#' is converted to %23.

**Comment by karelz** [comment]
@pulkjr can you please provide a simplified repro? Which version of .NET Core do you use?

**Comment by pulkjr** [comment]
I'm executing this via PowerShell. 

```powershell
PS /git/NetAppDocs.HealthCheckRules> dotnet --version                                                                                                    
2.1.104

PS /git/NetAppDocs.HealthCheckRules> $PSVersionTable                                                                                                     

Name                           Value                                                                                                                                
----                           -----                                                                                                                                
PSVersion                      6.0.2                                                                                                                                
PSEdition                      Core                                                                                                                                 
GitCommitId                    v6.0.2                                                                                                                               
OS                             Darwin 17.4.0 Darwin Kernel Version 17.4.0: Sun Dec 17 09:19:54 PST 2017; root:xnu-4570.41.2~1/RELEASE_X86_64                        
Platform                       Unix                                                                                                                                 
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                                                              
PSRemotingProtocolVersion      2.3                                                                                                                                  
SerializationVersion           1.1.0.1                                                                                                                              
WSManStackVersion              3.0                                                                                                                                  

PS /git/NetAppDocs.HealthCheckRules> $ds = [System.Data.DataSet]::new()                                                                                  
PS /git/NetAppDocs.HealthCheckRules> $ds.ReadXmlSchema('/git/NetAppDocs.HealthCheckRules/Resources/Schema/NetAppDocs.NcController.Document#BuildDoc.xsd')                                                                                                                                                         
Exception calling "ReadXmlSchema" with "1" argument(s): "Could not find file '/git/NetAppDocs.HealthCheckRules/Resources/Schema/NetAppDocs.NcController.Document'."
At line:1 char:1
+ $ds.ReadXmlSchema('/git/NetAppDocs.HealthCheckRules/Resou ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
+ FullyQualifiedErrorId : FileNotFoundException
```
This does work though
```powershell
PS /Users/jpulk/git/NetAppDocs.HealthCheckRules> $ds.ReadXmlSchema([uri]::new('/git/NetAppDocs.HealthCheckRules/Resources/Schema/NetAppDocs.NcController.Document#BuildDoc.xsd').tostring())  
```

**Comment by karelz** [comment]
Does it fail also with C#?
```c#
    new System.Data.DataSet().ReadXmlSchema("a#b.xsd");
```

It works fine for me with .NET Core 2.0 on Win10.

**Comment by karelz** [comment]
Seems to be duplicate / similar to #25820

**Comment by krwq** [comment]
@karelz, seems to be the same as https://github.com/dotnet/corefx/issues/26210. Could you try 2.1 preview and see if it repros?

If still repros could you share `dotnet --info` output?

**Comment by rmkerr** [comment]
I don't think this is related to #26210 -- that issue is specific to URIs containing Unicode characters.

This is probably a duplicate of the issue @karelz mentioned, or at least very similar.

**Comment by krwq** [comment]
@pulkjr this should be already fixed in 2.1 preview. Please test and reopen if you still see the problem (please share exact paths causing the problems and output of `dotnet --info` - I expect it to be fixed though since we can see tests related to that passing).
